docs/adr/0035-refactor-payments-frontend.md

```diff
@@ -0,0 +1,101 @@
+
+- Status: proposed
+- Deciders: Ben Bangert, Lisa Chan, Barry Chen
+- Date: 2023-03-07
+
+## Context and Problem Statement
+
+The Subscription Platform frontend has become overly complex, making it inflexible and difficult to work with. This has resulted in longer development times on new features and in some cases left us unable to easily accommodate requests from relying parties.
+
+Additionally, as the app has grown, so have the number of network calls causing a reduction in app performance. The aim is to refactor the app to have one network call for each action.
+
+Finally, the frontend includes some business and application logic, which has made components and parts of the app unnecessarily complex. As a result the frontend is less flexible, and harder to change and maintain. To solve this, the Subscription Platform team wants to move all business and application logic to the backend.
+
+To address the problems briefly discussed above, and to meet current and future relying party and feature requests, refactoring of the existing Subscription Platform is necessary.
+
+## Decision Drivers
+
+- Faster development - This applies to both the refactoring necessary for SubPlat 3.0, as well as future relying party and feature requests
+- Easier routing - The SubPlat teams wants to perform more advanced routing to allow features such as, Relying Party specific checkout pages, A/B testing, etc,
+- GraphQL support - Improve app performance and responsivness by reducing the number of network requests
+- Reuse existing components - We have received generally favourable reviews of our existing frontend look and feel, and want to keep the appearance the same
+
+## Considered Options
+
+- A. React Framework - Next.js
+- B. React Framework - Gatsby.js
+- C. React, Typescript, without Redux
+- D. React, Typescript, with Redux
+- E. React Framework - Remix
+
+## Decision Outcome
+
+Chosen option: "A. React Framework - Next.js", because a significant amount of refactoring is necessary to accomplish the Subscription Platforms goals, now would be a good time to adopt a React framework and start from a mostly clean slate, since it partially or completely meets all of the decision drivers, especially the easier routing and faster development drivers.
+
+Regarding why Next.js is favored over Gatsby.js, before completing the ADR, the subscription platform team went through a prototyping project and found the two frameworks to be remarkably similar in capabilities. However due to Next.js's popularity, use by other teams in Mozilla and a team vote, the decision was made to pick Next.js.
+
+## Pros and Cons of the Options
+
+### A. React Framework - Next.js
+
+Use a React framework, Next.js.
+
+- Good, because it provides many benefits out of the box, without compromising flexibility, that would require additional libraries otherwise. (Routing, rendering options, image optimization, etc.)
+- Good, because of its routing simplicity and flexibility
```

Comment:
It's mentioned a couple of times that the routing is nicer and more flexible and even nicer for RPs. It could be worth noting what it offers specifically besides abstraction into a `pages/` directory?

It could also be a neutral point that NextJS is more opinionated than vanilla React. Nice for some things, might could limit us other times.

Reply:
I've updated the point as follows. I'd like to keep it in the Good column, since in this case it hits 2 of the decision drivers, Faster development and Easier routing.

```
Good, because of its routing simplicity and flexibility. The file-based routing is simple to understand as a developer, removes some boilerplate code, and easily allows changes to available routes. Although not as flexible as other libraries, it still provides the dynamic routes required for this refactor.
```

Reply:
👍 I think it's fine to leave the routing in the good column, my 2nd point was about NextJS being more opinionated than vanilla React (which we may not like in some scenarios), but maybe that's also all fine and we like that.

Reply:
Yup, I think in this case we do like that it's more opinionated. Although I did find it to still be flexible and still allows the features we are looking for.